src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1205,10 +1205,11 @@ void grpc_chttp2_complete_closure_step(grpc_chttp2_transport* t,
   }
   if (closure->next_data.scratch < CLOSURE_BARRIER_FIRST_REF_BIT) {
     if ((t->write_state == GRPC_CHTTP2_WRITE_STATE_IDLE) ||
-        !(closure->next_data.scratch & CLOSURE_BARRIER_MAY_COVER_WRITE)) {
+        !(closure->next_data.scratch & CLOSURE_BARRIER_MAY_COVER_WRITE) ||
```

Comment:
so.. run the closure immediately if there is an error (just thinking to myself).. though documentation might help

Reply:
Added a note.